File: public/models/webhook.ts
import { StringObject } from "@fider/services"

export interface WebhookData {
  name: string
  type: WebhookType
  status: WebhookStatus
  url: string
  content: string
  http_method: string
  http_headers: HttpHeaders
}

export interface Webhook extends WebhookData {
  id: number
}

export enum WebhookType {
  NEW_POST = "new_post",
  NEW_COMMENT = "new_comment",
  CHANGE_STATUS = "change_status",
  DELETE_POST = "delete_post",
}

export enum WebhookStatus {
  ENABLED = "enabled",
  DISABLED = "disabled",
  FAILED = "failed",
}

export type HttpHeaders = StringObject<string>

export interface WebhookTriggerResult {
  webhook: Webhook
  props: StringObject
  success: boolean
  url: string
  content: string
  status_code: number
  message: string
  error: string
}

export interface WebhookPreviewResult {
  url: PreviewedField
  content: PreviewedField
}

export interface PreviewedField {
  value: string
  message: string
  error: string
}

File: app/models/dto/webhook.go
package dto

import (
	"github.com/Spicy-Bush/fider-tarkov-community/app/models/entity"
	"github.com/Spicy-Bush/fider-tarkov-community/app/pkg/webhook"
)

type WebhookTriggerResult struct {
	Webhook    *entity.Webhook `json:"webhook"`
	Props      webhook.Props   `json:"props"`
	Success    bool            `json:"success"`
	Url        string          `json:"url"`
	Content    string          `json:"content"`
	StatusCode int             `json:"status_code"`
	Message    string          `json:"message"`
	Error      string          `json:"error"`
}

type WebhookPreviewResult struct {
	Url     PreviewedField `json:"url"`
	Content PreviewedField `json:"content"`
}

type PreviewedField struct {
	Value   string `json:"value,omitempty"`
	Message string `json:"message,omitempty"`
	Error   string `json:"error,omitempty"`
}
